packages/react/src/Button/Button.test.tsx

```diff
@@ -269,4 +269,10 @@ describe('Button', () => {
     // Ensure the button does not have the CSS class when block is false (default)
     expect(btnEl.classList).not.toContain('Button--block')
   })
+
+  it.each(['small', 'medium', 'large'])('has the correct class when size="%s"', size => {
```

Comment:
This is cool, though perhaps the way jest is doing the interpolation isn't as clear as it could be, given there's easier ways to do it?

For reference, [here's how we've done this kind of thing before for Box](https://github.com/primer/brand/blob/e7afdaf9e4da7e77ed384a6de63fe2a4f4c750a8/packages/react/src/Box/Box.test.tsx#L105).

If you export the sizes as an array out of the Button.tsx file too, you can use those in the react.mdx file's prop table too.

Reply:
Thank you for the feedback and link 🙌

I did start with something like this, which I'm happy to revert to.

```tsx
it('has the correct class size is specified', () => {
  for (const size of ButtonSizes) {
    const testId = `test-button-${size}`
    const {getByTestId} = render(
      <Button size={size} data-testid={testId}>
        Button
      </Button>,
    )

    expect(getByTestId(testId)).toHaveClass(`Button--size-${size}`)
  }
})
```

I ended up refactoring it down to the `it.each` as that negated the need to manage testIds to work around the fact that all the different sized buttons were rendered to the same screen.

More than happy to stick to the conventions we've already got though, I'll push the above change up shortly 🙂

Reply:
Thanks. Could you use `getAllByRole` to do assert multiple button instances?

Reply:
Updated in a275e45c02144aa9bc15525ffcc35c3c27828545